Feature/stepseq

riban-bw opened the pull request:
Merged master and tweaked to get on-screen buttons working. I have also removed "alpha" from the title in the main screen because I think we are past that stage. We could use "beta" if you want.
I suggest we merge this with master so that users can benefit from the significant changes since the version currently in master and to reduce the support overhead of users switching between incompatible versions.

jofemodo commented on zyngui/zynthian_gui_fileselector.py:
```diff
@@ -0,0 +1,192 @@
+#******************************************************************************
+# ZYNTHIAN PROJECT: Zynthian GUI
+# 
+# Zynthian GUI Step-Sequencer Class
```
Mmmmm ... i think we could choose a better description for this piece of code ;-)

jofemodo commented on zyngui/zynthian_gui_base.py:
```diff
@@ -412,26 +412,4 @@ def cb_topbar(self,event):
 		self.zyngui.zynswitch_defered('S',1)
 
 
-	def cb_keybinding(self, event):
```
I see you moved this to zynthian_gui and captured events from the top window.... this is completely OK for me ;-)

riban-bw replied:
Yes - we want to have consistent key capture across the application and putting this at the top level allows both a more consistent implementation and more uniform access to the binding.

jofemodo commented on zyngui/zynthian_gui_fileselector.py:
```diff
@@ -0,0 +1,192 @@
+# 
+# Zynthian GUI Step-Sequencer Class
+# 
+# Copyright (C) 2015-2020 Fernando Moyano <jofemodo@zynthian.org>
```
I can't remember to have contributed to this piece of code, but if you mean i "inspired" the code in some way, you can leave the line to rest  ;-)

jofemodo commented on zyngui/zynthian_gui_rename.py:
```diff
@@ -0,0 +1,248 @@
+#******************************************************************************
+# ZYNTHIAN PROJECT: Zynthian GUI
+# 
+# Zynthian GUI Step-Sequencer Class
```
Same applies here ;-)

jofemodo commented on zyngui/zynthian_gui_rename.py:
```diff
@@ -0,0 +1,248 @@
+# 
+# Zynthian GUI Step-Sequencer Class
+# 
+# Copyright (C) 2015-2020 Fernando Moyano <jofemodo@zynthian.org>
```
And here, of course ...

riban-bw commented:
I have moved the constants back into the sequencer codebase but this does introduce a circular import. I don't think this is an issue because I don't actually use any functions or class variables, just these constants in one direction. I have made changes to address issues highlighted by this round of code review.

I had forgotten that I had added the _technology preview_ of on-screen keyboard and file selector but if you are happy to include this then we can devise a way to reuse in other modules.
